### PR TITLE
Generate rootstrap definitions dynamically

### LINKED ISSUE #128: Choosing rootstrap
When executing the run or build command, flutter-tizen must select the root strap appropriately according to the device profile.

**As-is**

- mobile profile

```bash
$ flutter-tizen build tpk --device-profile mobile -v
...
[  +15 ms] executing: /home/boram/tizen-studio/tools/ide/bin/tizen build-native -a arm -C Release -c llvm-10.0 -r wearable-4.0-arm.flutter -e -lflutter_tizen_mobile
...
[+9125 ms] [PLATFORM] Wearable-4.0
                    [ARCHITECTURE] armel
                    [COMPILER] LLVM-10.0
                    [CONFIGURATION] Release
                    [SBI] prdefine options = 
                    [SBI] command : /bin/sh -c /usr/bin/make -f "/home/boram/tizen-studio/tools/ide/resources/native/Build/makefile" -r 
```

- tv profile

```bash
$ flutter-tizen build -v --devicePprofile tv
...
[  +16 ms] executing: /home/boram/tizen-studio/tools/ide/bin/tizen build-native -a arm -C Release -c llvm-10.0 -r wearable-4.0-arm.flutter -e -lflutter_tizen_t
...
[+8907 ms] [PLATFORM] Wearable-4.0
                    [ARCHITECTURE] armel
                    [COMPILER] LLVM-10.0
                    [CONFIGURATION] Release
                    [SBI] prdefine options = 
                    [SBI] command : /bin/sh -c /usr/bin/make -f "/home/boram/tizen-studio/tools/ide/resources/native/Build/makefile" -r 
```

**Additional related matters**
- As far as I know, TV profiles do not have official public native rootstraps. Should we use common rootstrap for tv profile?
- Tizen Studio provides extensions to support rootstraps for iot profiles.

### PR DESCRIPTION
**Pull request by swift-kim**
- Rewrite `TizenSdk.getFlutterRootstrap()` to generate rootstrap definitions dynamically. For example, the mobile rootstrap is required when building native apps/plugins for mobile devices. (Previously the builder used the wearable rootstrap if the profile name in `tizen-manifest.xml` was set to `common`.)
- Remove old rootstrap definitions.

Minor changes:
- Exit immediately if `tizen-manifest.xml` could not be found. `TizenManifest.parseFromXml()` never returns null.
- Remove `TizenManifest.profile` and rename `TizenManifest.profileName` with `TizenManifest.profile`.
- Move `getTizenBuildArch()` from `tizen_build_target.dart` to `tizen_sdk.dart`.

Fixes #128.

### CODE REVIEW

File: lib/tizen_sdk.dart [hunk 1]
```diff
@@ -114,61 +113,87 @@ class TizenSdk {
...
-    } else {
-      id = '$profile-${arch == 'x86' ? 'emulator' : 'device'}.core';
+    if (profile == 'common') {
+      // The headless profile is not supported.
+      profile = 'iot-headed';
+    }
+    if (profile == 'tv') {
+      // Use the wearable rootstrap because the TV native rootstrap is not
+      // publicly available.
+      profile = 'wearable';
```
Comment (WonyoungChoi) [inline lib/tizen_sdk.dart:130]: How about providing any information to user when the requested profile or arch is changed?

File: lib/tizen_sdk.dart [hunk 2]
```diff
@@ -114,61 +113,87 @@ class TizenSdk {
...
 
-    // Tizen SBI reads rootstrap definitions from this directory.
-    final Directory pluginsDir = toolsDirectory
-        .childDirectory('smart-build-interface')
-        .childDirectory('plugins');
+    String type = arch == 'x86' ? 'emulator' : 'device';
+    if (arch == 'arm64') {
+      // The arm64 build is only supported by iot-headed-6.0+ rootstraps.
+      profile = 'iot-headed';
+      if ((double.tryParse(apiVersion) ?? 0) < 6.0) {
```
Comment (bbrto21) [inline lib/tizen_sdk.dart:137]: When API version is wrong, how about warning users?

**Review by bbrto21 — APPROVED** [review]
LGTM!

File: lib/tizen_sdk.dart (hunk 1 above)
Comment (swift-kim) [reply lib/tizen_sdk.dart]: Done!

**Comment by swift-kim** [comment]
@xuelian-bai In order to use in-house TV rootstraps, simply

1. Unzip `rootstraps.zip` into the `tizen-studio` directory,<p>
   ```sh
   ~$ unzip rootstraps.zip
   ```
2. Change the package API version to `5.0` (depends on the rootstrap you use) in `tizen-manifeset.xml`, and then
3. Use `flutter-tizen run` or `flutter-tizen build tpk --device-profile tv` to build an app.

**Review by HakkyuKim — APPROVED** [review]
LGTM

**Pull request merged**